### DISCUSSION TITLE
Treat a volume as a unified entity during reading when it's comprised of multiple chapters

### DISCUSSION THREAD

**Discussion by newyorker123** [Kareadita/Kavita#2897] [Ideas]
### Idea Description

When you purchase a digital manga volume from a website, their e-reader usually provides the following functionalities:

1. A progress bar that shows the total page numbers of the volume, allowing you to jump to any page within the volume.
2. A table of contents panel that lists each chapter name, allowing you to jump to any chapter within the volume.

In Kavita, I merged multiple chapters into a volume using the naming scheme "xxx Vol.01 Ch.1.cbz", "xxx Vol.01 Ch.2.cbz", etc. However, when you try to read volume 1 from the Volumes tab, there are the following problems:

1. The progress bar is only for chapter 1. If you want to jump to a middle page in the volume, you have to exit the reader first and select the chapter in the Chapters tab. Alternatively, you have to manually go through multiple chapters.

   It would be better to have a progress bar for the entire volume when you actually want to read the volume, instead of just a chapter.
2. If 1 is implemented, it would be nice to have a table of contents for the current volume.

I believe these improvements will make Kavita better simulate the experience of reading a volume of manga.

### Idea Category

Feature Enhancement

### Duration of Using Kavita

7 months

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by LtSquigs** [comment]
Commenting because I have a similar problem/feature request that I think could be addressable by the same solution as this feature request.

Like the original requestor, I also buy a lot of digital volumes of manga to store on Kavita, and occasionally want to be able to quickly browse the chapters of those volumes. However, it would be nice to be able to do so without needing to break the volume up into individual chapter files as the original requester has done. Rather it would be useful if Kavita could create "virtual" chapters from the metadata information in a volume file, in much the same way it makes "virtual" volumes from multiple chapter files by using the name of the file.

For the CBZ format this could maybe be done through the `Pages.Bookmark` field in `ComicInfo.xml` by treating Bookmarks that match a parsing pattern (like `Chapter XX` or `Ch. XX`) as a Chapter. For ePUBs perhaps the entries in the `toc.nsx` could be parsed for similar naming schemes.

This would be useful for achieving the same results as the original request while allowing the storage to still be kept to the volume level (and allow the experience to work with ePUBs).
